outscale/resource_outscale_keypair_test.go

```diff
@@ -105,32 +105,21 @@ func testAccCheckOutscaleOAPIKeyPairDestroy(s *terraform.State) error {
 			}).Execute()
 
 			if err != nil {
-				if strings.Contains(err.Error(), "RequestLimitExceeded:") {
-					return resource.RetryableError(err)
-				}
-				return resource.NonRetryableError(err)
+				return utils.CheckThrottling(err)
 			}
-
-			return resource.RetryableError(err)
+			return nil
 		})
 
+		if err != nil {
+			return err
+		}
 		if err == nil {
 			if len(resp.GetKeypairs()) > 0 {
 				return fmt.Errorf("still exist")
 			}
 			return nil
 		}
-
-		// Verify the error is what we want
-		ec2err, ok := err.(awserr.Error)
-		if !ok {
-			return err
-		}
-		if ec2err.Code() != "InvalidOAPIKeyPair.NotFound" {
-			return err
```

Comment:
We change the behaviour ?

Reply:
We do not check that ?

Reply:
Error was already returned
But you are right, I will add this condition before return
Thanks,